app/utils/utils.ts

```diff
@@ -17,18 +18,19 @@ export const currentDate = () => {
 };
 
 // from https://www.jacobparis.com/content/remix-pagination
-export function setNewSearchParamsString(
+export function setNewSearchParams(
   searchParams: URLSearchParams,
-  changes: Record<string, string | number | undefined>,
+  changes: Record<string, string | number | undefined | null> | AdminParams,
```

Comment:
I see a few instances where we [union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html) `Record<string, string>` with `AdminParams`. My perception is we're doing this because `AdminParams` are incompatible with `Record<string, string>`. *(And also because district types are an odd case. details on how to resolve this case at the end of the list of suggestions.)* This is because AdminParams have the shape `Record<string, string | null>`. The `null` makes it incompatible.

The union does resolve this conflict. However, I think it obfuscates the intent of the typing. If we create a new type, I think we can make it compatible with AdminParams and clarify the intended types.

1. New `SearchParamChanges` type:
```typescript
export type SearchParamChanges = Record<string, string | number | null | undefined> 
``` 

2. We can then use this type in the high-level functions:
- `app/utils/utils.ts line 21`
```typescript
export function setNewSearchParams(
   searchParams: URLSearchParams
   changes: SearchParamChanges
)
```
- `app/root.tsx line 172`
```typescript
const updateSearchParams = (
    nextSearchParams: SearchParamChanges,
```

3. And then in the dropdowns, AdminParams should now be compatible with `SearchParamChanges` and we can specify that the dropdown params must meet the `AdminParams` criteria.
- `app/components/AdminDropdown/BoroughDropdown.tsx line 8`
```typescript
setAdminParams: (value: AdminParams) => void;
```
- and all the other dropdowns, too.
- Adding [these value checks](https://github.com/NYCPlanning/ae-cp-map/pull/108/files#diff-4c2ea46ba29c278a05289ff73d08b954319ea5e8950e56c8f226335791a3ff75) to the district type dropdown, to ensure the new value conforms to the AdminParams type.

Reply:
Mmmm, I'm trying this myself and it's giving me a hard time 🙃

Reply:
Yeah so admittedly, I popped the union in there to just get the types to work lol But-- what I had in mind was to make it so that we didn't write functions around only admin params while still having the compact and readable benefit `AdminParams` provide in the dropdowns. I agree with having _something_ like SearchParamChanges like maybe even keeping `Record<string, string | number | null | undefined>` as is instead of a new type?

Reply:
Okay, it's seems to be from mixing `interfaces` and `types`. Changing `AdminParams` from an `interface` to a type seems to resolve the typing problems. I'm updating the above comment to include this change.

Reply:
Great, updated with `SearchParamChanges`